### PR TITLE
(SIMP-9239) Add GHA workflows to rubygem-simp-cli

### PR DESCRIPTION
**Pull request by op-ct**
This patch adds standardized GitHub Actions to provide GitHub-based
CI/CD workflows.  Among this first batch are:

* PR GLCI - Auto-trigger corresponding GLCI (if actor has write perms)
* Manual: PR GLCI - Team members can trigger corresponding GLCI for a PR
* PR GLCI Cleanup - Clean up any auto-triggered GLCI pipelines/branches
* PR Test Matrix

This patch also enforces the standardized asset baseline using
simp/puppetsync, and may apply other updates to ensure conformity.

CI: SKIP MATRIX
[SIMP-9376] #close
[SIMP-9239] #comment Standardized GHA + assets in rubygem-simp-cli

[SIMP-9376]: https://simp-project.atlassian.net/browse/SIMP-9376
[SIMP-9239]: https://simp-project.atlassian.net/browse/SIMP-9239

### CODE REVIEW

**Comment by op-ct** [comment]
This includes a workflow to deploy to rubygems.org.  We used to do this with simp-cli early on, but stopped—is there any value in resuming it now?

**Comment by lnemsick-simp** [comment]
I do not think it should be deployed as a RubyGem.  Its RPM spec file has dependencies that are essential and render the code useless of absent.

File: .github/workflows/tag_deploy_rubygem.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: ${{ steps.tag-check.outputs.annotation_title }}
+          body_path: /tmp/annotation.body
+          draft: false
+          prerelease: false
+
+  deploy-rubygem:
```
Comment (lnemsick-simp) [inline .github/workflows/tag_deploy_rubygem.yml:117]: IMHO this should be omitted.

File: .travis.yml [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-    ###    * https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
-    ###    * https://simp-project.atlassian.net/browse/SIMP-8703
-    ###
-    ###  - stage: check
-    ###    name: 'Syntax, style, and validation checks'
-    ###    rvm: 2.4.9
-    ###    script:
-    ###      - bundle exec rake pkg:compare_latest_tag
-    ###      - bundle exec rake pkg:create_tag_changelog
-    ###      - bundle exec rake pkg:gem
```
Comment (lnemsick-simp) [inline .travis.yml:44]: The pkg:gem check  is not in the .gitlab-ci.yml.  It needs to be.

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -81,7 +81,7 @@ variables:
 
 # To avoid running a prohibitive number of tests every commit,
 # don't set this env var in your gitlab instance
-.only_with_SIMP_FULL_MATRIX: &only_with_SIMP_FULL_MATRIX
+.with_SIMP_ACCEPTANCE_MATRIX_LEVEL_3: &with_SIMP_ACCEPTANCE_MATRIX_LEVEL_3
   only:
     variables:
       - $SIMP_FULL_MATRIX == "yes"
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml:87]: does this make sense or was this a bad merge?

**Review by lnemsick-simp — APPROVED** [review]

**Pull request merged**